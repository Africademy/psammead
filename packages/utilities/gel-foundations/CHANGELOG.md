# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
