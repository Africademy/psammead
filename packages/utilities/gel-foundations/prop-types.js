module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
