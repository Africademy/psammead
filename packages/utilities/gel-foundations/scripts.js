/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
