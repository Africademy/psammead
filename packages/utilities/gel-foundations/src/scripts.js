/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
