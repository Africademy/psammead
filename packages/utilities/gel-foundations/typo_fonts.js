module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
