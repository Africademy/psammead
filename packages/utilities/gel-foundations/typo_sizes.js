module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
