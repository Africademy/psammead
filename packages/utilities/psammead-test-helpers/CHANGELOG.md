# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
