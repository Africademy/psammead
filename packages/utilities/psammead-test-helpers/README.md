# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactjs.org/docs/how-to-contribute.html#your-first-pull-request)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install @bbc/psammead-test-helpers --save
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
