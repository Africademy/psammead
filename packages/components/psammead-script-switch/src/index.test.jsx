import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { render } from '@testing-library/react';
import { latin } from '@bbc/gel-foundations/scripts';
import ScriptSwitch from './index';

describe('ScriptSwitch', () => {
  const props = {
    service: 'news',
    href: 'https://www.bbc.co.uk/news',
    script: latin,
  };

  shouldMatchSnapshot(
    'should render correctly',
    <ScriptSwitch {...props}>Lat</ScriptSwitch>,
  );

  it('should render a link', () => {
    const { container } = render(<ScriptSwitch {...props}>Lat</ScriptSwitch>);
    const links = container.querySelectorAll('a');

    expect(links).toHaveLength(1);
    expect(links[0]).toHaveProperty('href');
    expect(links[0].href).toEqual('https://www.bbc.co.uk/news');
    expect(links[0].textContent).toEqual('Lat');
  });
});
