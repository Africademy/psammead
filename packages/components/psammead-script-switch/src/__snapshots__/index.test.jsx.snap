// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptSwitch should render correctly 1`] = `
.c0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  display: inline-block;
  border: 0.0625rem solid #FFFFFF;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 1rem;
  height: 3rem;
}

.c0:hover {
  border-width: 0.25rem;
  line-height: 2.62rem;
  padding: 0 0.81rem;
}

.c0:focus {
  background-color: #FFFFFF;
  color: #222222;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    line-height: 3rem;
  }
}

@media (max-width:37.5rem) {
  .c0 {
    line-height: 2.5rem;
    height: 2.5rem;
    padding: 0 0.75rem;
  }

  .c0:hover {
    line-height: 2.1rem;
    padding: 0 0.56rem;
  }
}

<a
  class="c0"
  href="https://www.bbc.co.uk/news"
>
  Lat
</a>
`;
